nifi-commons/nifi-sensitive-property-provider/src/main/java/org/apache/nifi/properties/AbstractSensitivePropertyProvider.java

```diff
@@ -47,4 +47,34 @@ public String getName() {
     public String getIdentifierKey() {
         return getProtectionScheme().getIdentifier();
     }
+
+    @Override
+    public String protect(final String unprotectedValue, final ProtectedPropertyContext context) throws SensitivePropertyProtectionException {
+        return protect(unprotectedValue);
+    }
+
+    @Override
+    public String unprotect(final String protectedValue, final ProtectedPropertyContext context) throws SensitivePropertyProtectionException {
+        return unprotect(protectedValue);
+    }
+
+    /**
+     * Allows implementing classes to provide an implementation that does not require a context.
+     * @param unprotectedValue The unprotected value
+     * @return The protected value
+     * @throws SensitivePropertyProtectionException If there was a problem protecting the value
+     */
+    protected String protect(final String unprotectedValue) throws SensitivePropertyProtectionException {
+        return null;
+    }
+
+    /**
+     * Allows implementing classes to provide an implementation that does not require a context.
+     * @param protectedValue The protected value
+     * @return The unprotected value
+     * @throws SensitivePropertyProtectionException If there was a problem unprotecting the value
+     */
+    protected String unprotect(final String protectedValue) throws SensitivePropertyProtectionException {
+        return null;
+    }
```

Comment:
Is there a reason for implementing these methods and returning `null` as opposed to marking them as `abstract`? It seems like having a default implementation that returns `null` could cause some confusion in other implementations.

Reply:
My thought was that an SPP would implement either the 1-arg or 2-arg version.  If this was abstract, SPPs that intended to only implement the 2-arg version would be forced to provide a no-op implementation of the 1-arg version.

Reply:
Thanks for the clarification, that is helpful. The concept makes sense given that some providers will use the context and some will not, however, it seems like a different approach may be necessary.

Providers that use the context parameter could implement the interface directly, whereas providers that do not need the context could extend the abstract class. In light of the existing functionality in this class, perhaps some refactoring along that line may be necessary, or perhaps two different abstract classes would be helpful. Either way, adjusting the implementation would make it clearer which approach a particular provider implements.

Reply:
On further consideration, due to the potential complexity of clearly distinguishing between 1-arg and 2-arg versions of these methods, I decided to revert the 1-arg change.  My thinking is that it will be simpler in the long run for all SPPs to use the 2-arg methods and each SPP implementation can decide whether it wants to use the context or not.